tests/linalg_test.py

```diff
@@ -42,6 +42,29 @@
 float_types = [np.float32, np.float64]
 complex_types = [np.complex64, np.complex128]
 
+
+# Used in tests for det and lu
+singular_mats = [
```

Comment:
Nit: I think it’s a best practice to keep data for data dependent tests defined inside or directly adjacent to the test method?

Reply:
I can move the non-square matrices, but `singular_mats` is used by two tests (for both determinants and LU decomposition). What's the best practice there?

Reply:
I moved nonsquare_singular_mats.

Reply:
The linter keeps complaining now and the tests are failing. I've reverted back to the way it was - at least that worked.